Thank you for stepping forward to help with this issue @07jasjeet 🙇 
My plan is to fully review and test this PR by the end of the week.

I've started by triggering the CI tests and noticed the following minor checkstyle issues.

```
> Task :WordPress:checkstyle
[ant:checkstyle] [ERROR] /Users/antonis/git/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:18: 'androidx.activity.OnBackPressedCallback' should be separated from previous imports. [ImportOrder]
[ant:checkstyle] [ERROR] /Users/antonis/git/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:26: 'com.google.android.gms.common.ConnectionResult' should be separated from previous imports. [ImportOrder]
[ant:checkstyle] [ERROR] /Users/antonis/git/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:29: 'org.greenrobot.eventbus.EventBus' should be separated from previous imports. [ImportOrder]
```

You can find some information on how to [run those checks locally here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/coding-style.md#checkstyle)

--- a/WordPress/src/main/java/org/wordpress/android/ui/PrivateAtCookieRefreshProgressDialog.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/PrivateAtCookieRefreshProgressDialog.kt
@@ -7,7 +7,6 @@ import android.app.ProgressDialog
 import android.content.DialogInterface
 import android.os.Bundle
 import androidx.fragment.app.DialogFragment
-import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import org.wordpress.android.R
 
@@ -28,11 +27,10 @@ class PrivateAtCookieRefreshProgressDialog : DialogFragment() {
         return dialog != null && dialog!!.isShowing
     }
 
-    @Suppress("DEPRECATION")
     override fun onCancel(dialog: DialogInterface) {
         super.onCancel(dialog)
-        if (targetFragment is PrivateAtCookieProgressDialogOnDismissListener) {
-            (targetFragment as PrivateAtCookieProgressDialogOnDismissListener).onCookieProgressDialogCancelled()
+        if (parentFragment is PrivateAtCookieProgressDialogOnDismissListener) {
+            (parentFragment as PrivateAtCookieProgressDialogOnDismissListener).onCookieProgressDialogCancelled()
         } else if (activity is PrivateAtCookieProgressDialogOnDismissListener) {
             (activity as PrivateAtCookieProgressDialogOnDismissListener).onCookieProgressDialogCancelled()
         }
@@ -42,18 +40,14 @@ class PrivateAtCookieRefreshProgressDialog : DialogFragment() {
         const val TAG = "private_at_cookie_progress_dialog"
 
         fun showIfNecessary(fragmentManager: FragmentManager?) {
-            showIfNecessary(fragmentManager, null)
-        }
-
-        fun showIfNecessary(fragmentManager: FragmentManager?, targetFragment: Fragment?) {
             fragmentManager?.let {
                 val thisFragment = fragmentManager.findFragmentByTag(TAG)
-                if (thisFragment == null ||
-                    (thisFragment is PrivateAtCookieRefreshProgressDialog && !thisFragment.isDialogVisible())
-                ) {
+                if (thisFragment == null) {
                     val progressFragment = PrivateAtCookieRefreshProgressDialog()
-                    progressFragment.setTargetFragment(targetFragment, 0)
                     progressFragment.show(fragmentManager, TAG)
+                }
+                if (thisFragment is PrivateAtCookieRefreshProgressDialog && !thisFragment.isDialogVisible()) {
+                    thisFragment.show(fragmentManager, TAG)
                 }
             }
         }

--- a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
@@ -15,6 +15,7 @@ import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
 import android.widget.Toast;
+
 import androidx.activity.OnBackPressedCallback;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
@@ -23,9 +24,11 @@ import androidx.core.app.RemoteInput;
 import androidx.fragment.app.Fragment;
 import androidx.fragment.app.FragmentManager;
 import androidx.lifecycle.ViewModelProvider;
+
 import com.google.android.gms.common.ConnectionResult;
 import com.google.android.gms.common.GoogleApiAvailability;
 import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
 import org.greenrobot.eventbus.EventBus;
 import org.greenrobot.eventbus.Subscribe;
 import org.greenrobot.eventbus.ThreadMode;

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt
@@ -26,7 +26,6 @@ import android.webkit.WebView
 import android.widget.ImageView.ScaleType.CENTER_CROP
 import android.widget.ProgressBar
 import android.widget.TextView
-import android.widget.Toast
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.ListPopupWindow

--- a/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt
+++ b/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.kt
@@ -26,6 +26,7 @@ import android.webkit.WebView
 import android.widget.ImageView.ScaleType.CENTER_CROP
 import android.widget.ProgressBar
 import android.widget.TextView
+import android.widget.Toast
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.ListPopupWindow
@@ -39,7 +40,6 @@ import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
 import androidx.core.view.isInvisible
 import androidx.core.view.isVisible
-import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import androidx.fragment.app.viewModels
 import androidx.lifecycle.Lifecycle
@@ -904,7 +904,7 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
             is ReaderNavigationEvents.ReplaceRelatedPostDetailsWithHistory ->
                 replaceRelatedPostDetailWithHistory(postId = this.postId, blogId = this.blogId)
 
-            is ReaderNavigationEvents.ShowPostInWebView -> showPostInWebView(post, this@ReaderPostDetailFragment)
+            is ReaderNavigationEvents.ShowPostInWebView -> showPostInWebView(post)
             is ReaderNavigationEvents.ShowEngagedPeopleList -> {
                 ActivityLauncher.viewPostLikesListActivity(
                     activity,
@@ -1554,7 +1554,7 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
         null -> false
     }
 
-    private fun showPostInWebView(post: ReaderPost, fragment: Fragment) {
+    private fun showPostInWebView(post: ReaderPost) {
         readerWebView.setIsPrivatePost(post.isPrivate)
         readerWebView.setBlogSchemeIsHttps(UrlUtils.isHttps(post.blogUrl))
         readerProgressBar.visibility = View.VISIBLE
@@ -1567,7 +1567,7 @@ class ReaderPostDetailFragment : ViewPagerFragment(),
 
         // if the post is from private atomic site postpone render until we have a special access cookie
         if (post.isPrivateAtomic && privateAtomicCookie.isCookieRefreshRequired()) {
-            PrivateAtCookieRefreshProgressDialog.showIfNecessary(childFragmentManager, fragment)
+            PrivateAtCookieRefreshProgressDialog.showIfNecessary(childFragmentManager)
             requestPrivateAtomicCookie()
         } else if (post.isPrivateAtomic && privateAtomicCookie.exists()) {
             // make sure we add cookie to the cookie manager if it exists before starting render

